php/libraries/NDB_Caller.class.inc

```diff
@@ -361,8 +400,12 @@ class NDB_Caller
                 $_REQUEST['sessionID']
             );
         }
-        if (method_exists($menu, 'save')) {
+        if (method_exists($menu, 'save') && self::shouldCallSave()) {
```

Comment:
It would probably make sense to make this non-static, and move the method_exists check into shouldCallSave. (If the method doesn't exist, it should never be called..)